### ISSUE TITLE
False positive on MissingClosureParamType?

### ISSUE THREAD

**Issue by leocavalcante** [vimeo/psalm#2383]
```php
<?php declare(strict_types=1);

/**
 * @return Closure(int): string
 */
function foo(): Closure
{
    /**
     * @param int $bar
     * @psalm-param int $bar
     * @return string
     * @psalm-return string
     */
    return function ($bar) {
        return strval($bar);
    };
}
```
Not sure if I'm just annotating it wrong but this snippet erros with:
```
ERROR: MissingClosureParamType - index.php:14:22 - Parameter $bar has no provided type
    return function ($bar) {
```

**Comment by muglug** [comment]
You have to put the docblock directly before the closure (putting it above the return doesn’t work).

Like this:

```php
/**
 * @return Closure(int): string
 */
function foo(): Closure
{
    return
    /** @param int $bar */
    function ($bar) {
        return strval($bar);
    };
}
```

Alternatively just use an actual param type:
```php
/**
 * @return Closure(int): string
 */
function bar(): Closure
{
    return function (int $bar) {
        return strval($bar);
    };
}
```

In both cases, you can safely omit the `@return` - Psalm infers that from `strval`.

https://psalm.dev/r/06d3eae01f

**Comment by leocavalcante** [comment]
Thank you!
